app/controllers/concerns/sufia/files_controller_behavior.rb

```diff
@@ -268,5 +274,13 @@ def update_metadata_from_upload_screen
         @generic_file.relative_path = params[:relative_path] if params[:relative_path]
         @generic_file.on_behalf_of = params[:on_behalf_of] if params[:on_behalf_of]
       end
+
+      def add_file_to_collection
+        collection_id = params[:collection]
+        return if collection_id.nil? || collection_id == -1
+        collection = Collection.find(collection_id)
+        collection.member_ids = [@generic_file.id].concat(collection.member_ids)
+        ActiveFedora::SolrService.instance.conn.commit if collection.save
```

Comment:
This came from the code used by Dashboard -> My Files -> select file -> Add to Collection -> Update Collection, which runs code in Hydra:Collection - [Hydra::CollectionControllerBehavior](https://github.com/projecthydra/hydra-collections/blob/master/app/controllers/concerns/hydra/collections_controller_behavior.rb) #create (line 68) -> #add_members_to_collection (line 173) -> #after_create (line 53).  I wasn't able to use this code directly.  It was reproduced in Sufia.  Ideally, Hydra:Collections would have this code generalized out of the controller so it can be called in other situations where including the controller is not an option.  I wasn't looking at updating Hydra:Collections, but I can look at that if you like.